pkg/controller/endpointslice/topologycache/topologycache.go

```diff
@@ -263,6 +269,13 @@ func (t *TopologyCache) SetNodes(nodes []*v1.Node) {
 
 // HasPopulatedHints checks whether there are populated hints for a given service in the cache.
 func (t *TopologyCache) HasPopulatedHints(serviceKey string) bool {
+	t.lock.Lock()
+	defer t.lock.Unlock()
+
+	return t.hasPopulatedHintsLocked(serviceKey)
+}
+
+func (t *TopologyCache) hasPopulatedHintsLocked(serviceKey string) bool {
```

Comment:
the naming might be a bit misleading, what about `hasPopulatedHintsWithoutLock` or `hasPopulatedHintsThreadUnsafe`?

Reply:
Are you sure?  I see the naming pattern of `foo` (or `Foo`) that takes a lock and `fooLocked` that assumes a lock is held all over the Kubernetes codebase when I grep for `func .*Locked(`.  Here's one example out of dozens: https://github.com/kubernetes/kubernetes/blob/decf1e1a9b53a4f4b24bb780d6433e2d5e959d83/staging/src/k8s.io/client-go/tools/cache/heap.go#L188-L216

I see one match for `func.*WithoutLock`, but it actually takes a lock: https://github.com/kubernetes/kubernetes/blob/decf1e1a9b53a4f4b24bb780d6433e2d5e959d83/vendor/github.com/peterbourgon/diskv/diskv.go#L546-L551

I see only a couple matches for `func.*ThreadUnsafe`, e.g.: https://github.com/kubernetes/kubernetes/blob/decf1e1a9b53a4f4b24bb780d6433e2d5e959d83/pkg/kubelet/util/manager/watch_based_manager.go#L66-L82

In conclusion:

* `fooWithoutLock` seems to mean the opposite of what we want here. 
* `fooThreadUnsafe` is unusual but has precedent.
* `fooLocked` seems to be the prevailing pattern.  

However, if you strongly prefer "hasPopulatedHintsThreadUnsafe", I'll make the change.  Let me know if you indeed want me to do that.